src/reducers/scrollState.js

```diff
@@ -72,14 +71,18 @@ function scrollState(state = DEFAULT_STATE, action) {
 
       if (newProps.scrollTop && newProps.scrollTop !== oldProps.scrollTop) {
         state = scrollTo(state, newProps.scrollTop);
-        state = updateVisibleRows(state);
       }
 
       if (newProps.scrollToRow && newProps.scrollToRow !== oldProps.scrollToRow) {
         state = scrollToRow(state, newProps.scrollToRow);
-        state = updateVisibleRows(state);
       }
 
+      if (newProps.rowsCount !== state.rowsCount) {
+        state = updateRowCount(state, newProps);
+      }
+
+      state = updateVisibleRows(state);
```

Comment:
Hmm this is going to have performance issues. I tried to only call updateVisibleRows when something it depended on changed. 

Lets keep this in but add a TODO (and file a github issue) to clean it up.